src/main/java/org/alliancegenome/curation_api/controllers/crud/GraphQLController.java

```diff
@@ -0,0 +1,36 @@
+package org.alliancegenome.curation_api.controllers.crud;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.alliancegenome.curation_api.model.entities.Gene;
+import org.alliancegenome.curation_api.model.input.Pagination;
+import org.alliancegenome.curation_api.response.ObjectResponse;
+import org.alliancegenome.curation_api.response.SearchResponse;
+import org.alliancegenome.curation_api.services.GeneService;
+import org.apache.logging.log4j.plugins.Namespace;
+import org.eclipse.microprofile.graphql.Description;
+import org.eclipse.microprofile.graphql.GraphQLApi;
+import org.eclipse.microprofile.graphql.Query;
+
+import jakarta.inject.Inject;
+
+@GraphQLApi
+@Namespace("gene")
+@Description("Gene operations")
+public class GraphQLController {
+
+	@Inject GeneService geneService;
+	
+	@Query
+	public ObjectResponse<Gene> getGeneByIdentifier(String geneId) {
+		return geneService.getByIdentifier(geneId);
+	}
+
+	@Query
+	public SearchResponse<Gene> findGeneByParams(Integer page, Integer limit, Map<String, String> params) {
```

Comment:
Why do you not define the Map with Map<String, Object>? Then you don't need to convert it here.

Reply:
GraphQL does not support <String, Object> only <String, String>

Reply:
Ok.